core/src/test/java/org/elasticsearch/action/support/master/TransportMasterNodeActionTests.java

```diff
@@ -320,4 +322,35 @@ protected void processBeforeDelegationToMaster(Request request, ClusterState sta
             }
         }
     }
+
+    public void testMasterFailoverAfterStepDown() throws ExecutionException, InterruptedException {
+        Request request = new Request();
+        PlainActionFuture<Response> listener = new PlainActionFuture<>();
+
+        final Response response = new Response();
+
+        clusterService.setState(ClusterStateCreationUtils.state(localNode, localNode, allNodes));
+
+        new Action(Settings.EMPTY, "testAction", transportService, clusterService, threadPool) {
```

Comment:
can we increase the timeout on the request? if one runs a debugger the test may fail to retry because a timeout happens, making it confusing.